### PR TITLE
RavenDB-17955 - Debug Info Package takes a very long time to generate

### PR DESCRIPTION
**Pull request by shiranshalom**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17955

### Additional description

We failed to reproduce the bug; for now, we added more information for the debug info package (the execution time of each endpoint).

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs [hunk 1]
```diff
@@ -75,5 +75,23 @@ await using (var sw = new StreamWriter(entryStream))
                 await sw.FlushAsync();
             }
         }
+
+        public static async Task WriteExtraDebugInfoAsZipEntryAsync(Dictionary<string, TimeSpan> debugInfoTimeSpans, ZipArchive archive, string databaseName)
```
Comment (garayx) [inline src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs]: private

File: src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs [hunk 2]
```diff
@@ -75,5 +75,23 @@ await using (var sw = new StreamWriter(entryStream))
...
+        {
+            var entryName = databaseName == null ? "DebugInfo-ServerWide" : $"DebugInfo-ServerWide-{databaseName}";
```
Comment (garayx) [inline src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs]:
```suggestion
            var entryName = databaseName == null ? "DebugInfo-ServerWide" : $"DebugInfo-{databaseName}";
```

File: src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs [hunk 3]
```diff
@@ -75,5 +75,23 @@ await using (var sw = new StreamWriter(entryStream))
...
+            var entry = archive.CreateEntry($"{entryName}.txt");
+            entry.ExternalAttributes = ((int)(FilePermissions.S_IRUSR | FilePermissions.S_IWUSR)) << 16;
+
+            var sorted = debugInfoTimeSpans.OrderBy(o => o.Value);
```
Comment (garayx) [inline src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs]: lets use OrderByDescending

File: src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs (hunk 1 above)
Comment (haludi) [inline src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs]: I don't like the `Extra`.
It doesn't tell me about the content.
Maybe something like WriteDebugInfoMetadataAsZipEntryAsync for a generic manner
Or WriteDebugInfoTimesAsZipEntryAsync for specific manner

File: src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs (hunk 2 above)
Comment (shiranshalom) [reply src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs]: done

File: src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs (hunk 3 above)
Comment (shiranshalom) [reply src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs]: done

File: src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs (hunk 1 above)
Comment (shiranshalom) [reply src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs]: done

Comment (shiranshalom) [reply src/Raven.Server/ServerWide/DebugInfoPackageUtils.cs]: can't :-)

**Review by garayx — APPROVED** [review]

**Review by haludi — APPROVED** [review]

**Pull request merged**